e2e/deployers/discoveredapps.go

```diff
@@ -0,0 +1,89 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package deployers
+
+import (
+	"fmt"
+
+	"github.com/ramendr/ramen/e2e/util"
+	"github.com/ramendr/ramen/e2e/workloads"
+)
+
+type DiscoveredApps struct{}
+
+func (s DiscoveredApps) GetName() string {
+	return "Disapp"
+}
+
+func (s DiscoveredApps) Deploy(w workloads.Workload) error {
+	util.Ctx.Log.Info("enter Deploy " + w.GetName() + "/" + s.GetName())
+
+	name := GetCombinedName(s, w)
+	namespace := name
+
+	// create namespace in both dr clusters
+	if err := util.CreateNamespaceAndAddAnnotation(namespace); err != nil {
+		return err
+	}
+
+	// create pvc and deployment on dr1
```

Comment:
If we add a StatefulSet as another workload example then this deployer would change to accommodate that? Currently the deployer is tied to the deployment workload, this should be more generic.

For example:
- Deploy for discovered fetches a resource list from the Workload and deploys it to the target ManagedCluster
OR,
- Deploy for deployed uses kustomize to directly point to the workload base artifacts

As an example here:
  - CreateNamespaceAndAddAnnotation
  - w.GetDeployKustomization()
    - Returns a file path to a local kustomize file with the following contents (current assumption is Workload is of type deployment):
    ```
    resources:
    - https://raw.githubusercontent.com/$(GIT_URL)/$(BRANCH)/workloads/deployment/base/kustomization.yaml
    patches:
    - target:
        kind: PersistentVolumeClaim
        name: busybox-pvc
      patch: |-
        - op: replace
          path: /spec/storageClassName
          value: $(SCName)
    ```
  - DeployToCluster()
    - This is going to be tricky as we need to invoke a CLI to perform `kubectl apply -k <path-returned-by-w.GetDeploymentKustomization>`

With the above, we can deploy any workload in a discovered manner, and need to know about the actual resources that are present within the Workload and preserve the abstraction.

Reply:
@ShyamsundarR In the beginning of e2e, you asked us to not call cli command, so we manually created a few CRs (namespace, mcsb, placement, subscription) for subscription deployer, to replace "kubectl apply -k" command.

So are you sure you want to use kubectl cli for discovered apps deployer?

> If we add a StatefulSet as another workload example then this deployer would change to accommodate that?

yes, I planned to do such thing for discovered apps deployer. I will check the workload type, and then create different resources for each workload type, inside the deploy function. It is not elegant, but I do not see a better way.

Reply:
> @ShyamsundarR In the beginning of e2e, you asked us to not call cli command, so we manually created a few CRs (namespace, mcsb, placement, subscription) for subscription deployer, to replace "kubectl apply -k" command.
> 
> So are you sure you want to use kubectl cli for discovered apps deployer?

I do not see an easier way out at present. I tried looking at kustomize APIs and/or what internally `kubectl -k` does, which can serve as a code based way forward, but we can address that in a separate PR. IOW, CLI for this at present is fine.

> 
> > If we add a StatefulSet as another workload example then this deployer would change to accommodate that?
> 
> yes, I planned to do such thing for discovered apps deployer. I will check the workload type, and then create different resources for each workload type, inside the deploy function. It is not elegant, but I do not see a better way.

If we did the `kubectl -k` as above, we do not need to add any more code for different resource types. Which is what I am looking at, IOW for each workload type the deployer does not change, it calls the apply with kustomize for all.

Reply:
Updated code to use `kubectl apply -k` and `kubectl delete -k`